### PR TITLE
Convert to typescript

### PR DESCRIPTION
**Pull request by Page-**

### CODE REVIEW

File: src/types/color.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	websql: 'INTEGER',
+	odata: {
+		name: 'Self.Color',
+		complexType: `\
+<ComplexType Name="Color">
+	<Property Name="r" Nullable="false" Type="Edm.Int8"/>\
+	<Property Name="g" Nullable="false" Type="Edm.Int8"/>\
+	<Property Name="b" Nullable="false" Type="Edm.Int8"/>\
+	<Property Name="a" Nullable="false" Type="Edm.Int8"/>\
+</ComplexType>`,
```
Comment (zvin) [inline src/types/color.ts]: NIT: this could looks nicer with https://www.npmjs.com/package/outdent

File: package.json [hunk 2]
```diff
@@ -4,17 +4,22 @@
   "description": "SBVR type definitions.",
   "main": "out",
   "scripts": {
-    "lint": "resin-lint src test",
+    "lint": "resin-lint test && resin-lint --typescript src",
     "pretest": "npm run lint && npm run prepare",
     "test": "mocha",
     "prepublish": "require-npm4-to-publish",
-    "prepare": "coffee --output out --compile src"
+    "prepare": "npx tsc",
```
Comment (zvin) [inline package.json]: Why is npx needed? Isn't `tsc` already installed?

File: package.json [hunk 3]
```diff
@@ -27,8 +32,24 @@
     "chai-as-promised": "^7.1.1",
     "chai-datetime": "^1.5.0",
     "coffeescript": "^2.3.2",
-    "mocha": "^5.2.0",
+    "husky": "^1.3.1",
+    "lint-staged": "^8.1.5",
+    "mocha": "^6.0.2",
+    "prettier": "^1.16.4",
```
Comment (zvin) [inline package.json]: NIT: prettier is already installed by resin-lint

File: package.json [hunk 4]
```diff
@@ -27,8 +32,24 @@
     "chai-as-promised": "^7.1.1",
     "chai-datetime": "^1.5.0",
     "coffeescript": "^2.3.2",
```
Comment (zvin) [inline package.json]: There's no coffee files left

File: src/type-utils.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
+import * as _ from 'lodash';
+import * as Promise from 'bluebird';
```
Comment (zvin) [inline src/type-utils.ts]: NIT: I'd call this `Bluebird` (but it's like this in all files)

File: src/types/case-insensitive-text.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+import * as _Promise from 'bluebird';
```
Comment (zvin) [inline src/types/case-insensitive-text.ts]: Why is this import needed?

File: src/types/concept-type.ts [hunk 7]
```diff
@@ -0,0 +1,18 @@
+import * as _Promise from 'bluebird';
```
Comment (zvin) [inline src/types/concept-type.ts]: Why is this import needed?

File: src/types/file.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
+import * as _Promise from 'bluebird';
```
Comment (zvin) [inline src/types/file.ts]: Why is this import needed?

File: src/types/foreign-key.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
+import * as _Promise from 'bluebird';
```
Comment (zvin) [inline src/types/foreign-key.ts]: Why is this import needed?

File: src/types/integer.ts [hunk 10]
```diff
@@ -0,0 +1,18 @@
+import * as _Promise from 'bluebird';
```
Comment (zvin) [inline src/types/integer.ts]: Why is this import needed?

File: package.json (hunk 2 above)
Comment (Page-) [reply package.json]: It probably just came from the fact I normally run tsc with `npx tsc` so typed it from habit

File: package.json (hunk 3 above)
Comment (Page-) [reply package.json]: Yeah but we also use it directly so I want to be explicit about it being a dependency

File: package.json (hunk 4 above)
Comment (Page-) [reply package.json]: The tests are still in coffeescript

File: src/types/case-insensitive-text.ts (hunk 6 above)
Comment (Page-) [reply src/types/case-insensitive-text.ts]: Because we generate declaration files and `validate` returns a bluebird promise, so we need this unused import to have the bluebird types available for the declaration file, it's weird but it breaks without it

File: package.json (hunk 2 above)
Comment (Page-) [reply package.json]: fixed
